src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java

```diff
@@ -66,69 +64,107 @@ public List<Content> getContentsWithUserProgressData(UUID userId,
+                                metadata {
+                                    name
+                                    tagNames
+                                    suggestedDate
+                                    type
+                                    chapterId
+                                    rewardPoints
+                                }
+                                assessmentMetadata {
+                                    skillType
+                                    skillPoints
+                                }
+                                progressDataForUser(userId: $userId) {
+                                    userId
+                                    contentId
+                                    learningInterval
+                                    nextLearnDate
+                                    lastLearnDate
+                                    log {
+                                        timestamp
+                                        success
+                                        correctness
+                                        hintsUsed
+                                        timeToComplete
+                                    }
                                 }
                             }
                         }
                     }
                                     
                     """;
 
-            log.info("Sending contentsByChapterIds query to course service with chapterIds {}", chapterIds);
+            log.info("Sending contentsByChapterIds query to course service with chapterId " + chapterId);
 
             // we must use media content here because the content type is an interface
             // that cannot be used for deserialization
-            List<ContentWithUserProgressData[]> result = graphQlClient.document(query)
+            List<GenericAssessmentResponse> result = graphQlClient.document(query)
                     .variable("userId", userId)
-                    .variable("chapterIds", chapterIds)
-                    .retrieve("contentsByChapterIds")
-                    .toEntityList(ContentWithUserProgressData[].class)
+                    .variable("chapterIds", List.of(chapterId))
+                    .retrieve("contentsByChapterIds[0]")
+                    .toEntityList(GenericAssessmentResponse.class)
                     .retry(RETRY_COUNT)
                     .block();
 
             if (result == null) {
-                return List.of();
+                return Collections.emptyList();
             }
-            return result.stream()
-                    .flatMap(Arrays::stream)
-                    .map(ContentWithUserProgressData::toContent)
-                    .toList();
+            return result;
         } catch (Exception e) {
             throw new ContentServiceConnectionException("Error while fetching contents from content service", e);
         }
     }
 
+    public static class GenericAssessmentResponse {
```

Comment:
Why not make it a record? Or at least use `@Getter` and `@Setter`?

Reply:
Originally I used inheritance which doesn't work with records I think. I changed it to use inheritance now so the class stays, but you are obviously correct that i can at least use the annotations